Lib/queue.py

```diff
@@ -275,6 +275,13 @@ def _put(self, item):
     def _get(self):
         return self.queue.popleft()
 
+    def __iter__(self):
+        while True:
```

Comment:
I would move the`while True` inside the `try/except block`. This will keep the inner loop as compact and efficient as possible.

In theory, CPython has zero-cost try-blocks but that hasn't always been true and is not always guaranteed to be true. It is also not true for other implementations (MicroPython and CircuitPython for example).